src/componentsNext/Pagination/__tests__/describePagination.test.ts

```diff
@@ -0,0 +1,137 @@
+import { describe, expect, test } from "vitest";
+
+import describePagination from "../describePagination";
+
+describe("describePagination", () => {
+  describe("it describes a simple 10 page pagination", () => {
+    test("page 1 / 10", () => {
+      const pagination = describePagination({ pages: 10, currentPage: 1 });
+      expect(pagination).toStrictEqual([1, 2, "...", 9, 10]);
+    });
+
+    test("page 2 / 10", () => {
+      const pagination = describePagination({ pages: 10, currentPage: 2 });
+      expect(pagination).toStrictEqual([1, 2, 3, "...", 9, 10]);
+    });
+
+    test("page 5 / 10", () => {
+      const pagination = describePagination({ pages: 10, currentPage: 5 });
+      expect(pagination).toStrictEqual([1, 2, "...", 4, 5, 6, "...", 9, 10]);
+    });
+
+    test("page 6 / 10", () => {
+      const pagination = describePagination({ pages: 10, currentPage: 6 });
+      expect(pagination).toStrictEqual([1, 2, "...", 5, 6, 7, "...", 9, 10]);
+    });
+
+    test("page 7 / 10", () => {
+      const pagination = describePagination({ pages: 10, currentPage: 7 });
+      expect(pagination).toStrictEqual([1, 2, "...", 6, 7, 8, 9, 10]);
+    });
+
+    test("page 8 / 10", () => {
+      const pagination = describePagination({ pages: 10, currentPage: 8 });
+      expect(pagination).toStrictEqual([1, 2, "...", 7, 8, 9, 10]);
+    });
+    test("page 9 / 10", () => {
+      const pagination = describePagination({ pages: 10, currentPage: 9 });
+      expect(pagination).toStrictEqual([1, 2, "...", 8, 9, 10]);
+    });
+
+    test("page 10 / 10", () => {
+      const pagination = describePagination({ pages: 10, currentPage: 10 });
+      expect(pagination).toStrictEqual([1, 2, "...", 9, 10]);
+    });
+  });
+
+  describe("configure neighbours", () => {
+    test("by default it is configured have 1 neighbour", () => {
+      const defaultPagination = describePagination({
+        pages: 10,
+        currentPage: 1,
+      });
+
+      const oneNeighbourPagination = describePagination({
+        pages: 10,
+        currentPage: 1,
+        neighbours: 1,
+      });
+      expect(defaultPagination).toStrictEqual(oneNeighbourPagination);
+    });
+
+    test("increasing neighbours will eventually get rid of the ...", () => {
```

Comment:
A very minor wording suggestion: Should we use "ellipsis" to describe the dots?

We could also use the ellipsis character `…` everywhere, instead of three dots in the code. Technically this is more correct, but I don't think it matters a lot. Also, the ellipsis character is a bit wider when rendered in the DOM, so we would need to update the styling as well.

Reply:
I think this is a good idea and I actually would change it to the `…` sign as you suggested (and I would not worry about the spacing)